app/cdap/components/WidgetSVG/utils.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  fetchConnectionDetails,
+  fetchConnectors,
+  getCategoriesToConnectorsMap,
+} from 'components/Connections/Create/reducer';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import WidgetSVG from 'components/WidgetSVG';
+import {
+  IConnectorTypesWithSVG,
+  IConnectorDetailsPayload,
+  IConnectorTypes,
+} from 'components/WidgetSVG/types';
+import { importDatasetIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/importDatasetIcon';
+import React from 'react';
+
+export const getWidgetData = async () => {
+  const connectorTypes: IConnectorTypesWithSVG[] = await fetchConnectors();
+  const connectorsTypesData: IConnectorTypes[] = [];
+  const IConnectionWithConnectorType: IConnectorTypesWithSVG[] = [];
+  const allConnectorsPluginProperties: Map<
+    string,
+    IConnectorDetailsPayload[]
+  > = getCategoriesToConnectorsMap(connectorTypes);
+  const connectionPayload: IConnectorDetailsPayload[] = [];
+  allConnectorsPluginProperties?.forEach((eachProperty) => {
+    if (eachProperty.length) {
+      eachProperty.map((item) => {
```

Comment:
replace the map with forEach if you are not returning or getting an array out of it

Reply:
done